File: java/doit_java/Chapter7/src/standalone/Student.java
/*
나 혼자 코딩!
객체 배열 만들어 활용하기
학생 클래스 Student를 만들고 멤버 변수로 studentId, name을 선언합니다. showStudentInfo()
메서드를 만들어 studentId와 name 값을 출력합니다. 그리고 StudentArray 클래스에서
Student 3명의 배열을 만들고 Student를 생성하여 저장한 후 for문을 사용하여 Student 정보를
출력합니다. 출력 결과가 다음과 같도록 Studnet와 StudentArray 클래스를 만들어 보세요.
 */
package standalone;

public class Student {
    private int studentId;
    private String studentName;

    public Student() {}

    public Student(int studentId, String studentName) {
        this.studentId = studentId;
        this.studentName = studentName;
    }

    public int getStudentId() {
        return studentId;
    }

    public void setStudentId(int studentId) {
        this.studentId = studentId;
    }

    public String getStudentName() {
        return studentName;
    }

    public void setStudentName(String studentName) {
        this.studentName = studentName;
    }

    public void showStudentInfo() {
        System.out.println(studentId + ", " + studentName);
    }
}

File: python/p_oop2/chap06/exercise/student.py
from book import Book

class Student:
    def __init__(self, student_id, student_name):
        self.student_id = student_id
        self.student_name = student_name
        self.book_list = []  # 책에 대한 객체 정보를 저장하기 위한 리스트

    def add_book(self, name, author):
        book = Book(name, author)
        self.book_list.append(book)  # 책 리스트에 읽은 책의 이름과 저자를 객체로 저장

    def show_student_info(self):
        print(self.student_name + " 학생이 읽은 책은 :", end = ' ')
        for book in self.book_list:
            print(book.get_title(), end = ' ');
        print("입니다.")
